BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -137,7 +141,11 @@ uint8_t BSP_CAN_Read(uint32_t *id, uint8_t *data) {
 void CAN1_RX0_IRQHandler(void)
 {
     CAN_Receive(CAN1, CAN_FIFO0, &RxMessage);
-
+    for(int i = 0; i < 8; i++){
+        CANFIFO[CANFIFOHEAD][i] = RxMessage.Data[i]; //store data in FIFO
```

Comment:
If you are incrementing the fifo head each time, this will not work as you expect.

Reply:
I think I fixed this